doc/_scripts/gen_boards_catalog.py

```diff
@@ -203,6 +204,10 @@ def run_twister_cmake_only(outdir):
         "--outdir", str(outdir),
     ]
 
+    if vendor_filter:
+        for vendor in vendor_filter:
+            twister_cmd += ["--vendor", vendor]
```

Comment:
I think I'd rather have it be used in lieu of `--all`, when a filter has been provided, as it supersedes it anyway